ironfish/src/network/peerNetwork.ts

```diff
@@ -1066,6 +1071,11 @@ export class PeerNetwork {
       : (from: BlockHeader) => this.chain.iterateTo(from)
 
     for await (const header of iterationFunction(from)) {
+      remainingLookups -= 1
+      if (remainingLookups === 0) {
```

Comment:
I think there's an off-by-one here -- e.g. if remainingLookups starts at 1, it'll immediately decrement to 0 and break. Might be worth a test, if it's easy to write one.

Reply:
Good call - I had been avoiding trying to figure out how to test the max given it was a const in another file, but decided to just figure it out (tldr: not hard, just mockModule)